### PR TITLE
[Unit & Bullet] Missile tank

### PR DESCRIPTION
**Pull request by unprintable123**
添加了追踪导弹（以及测试用坦克）
也许之后会补一下导弹外观？

### CODE REVIEW

**Review by LazyJazz — DISMISSED** [review]
Nice and clean. LGTM!

File: src/battle_game/core/units/missile_tank.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  auto player = game_core_->GetPlayer(player_id_);
+  if (player) {
+    auto &input_data = player->GetInputData();
+    auto diff = input_data.mouse_cursor_position - position_;
+    if (glm::length(diff) < 1e-4) {
+      turret_rotation_ = rotation_;
+    }
+    turret_rotation_ = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/missile_tank.cpp:130]: Inherit from unit::Tank if you are using the very same code.

File: src/battle_game/core/units/missile_tank.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      turret_indices.push_back(precision + 1 + 1);
+      turret_indices.push_back(precision + 1 + 2);
+      turret_indices.push_back(precision + 1 + 1);
+      turret_indices.push_back(precision + 1 + 2);
+      turret_indices.push_back(precision + 1 + 3);
+      missle_tank_turret_model_index =
+          mgr->RegisterModel(turret_vertices, turret_indices);
+    }
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/missile_tank.cpp:71]: These too.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

**Comment by unprintable123** [comment]
改好了

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request merged**